### ISSUE TITLE
Repeated build errors M-store pro 3.9.1

### ISSUE THREAD

**Issue by smilechaitu** [inspireui/support#761]
My issue: when I am tried to build android app again after Gradle upgrade in latest version of yours. it gives error as per this screenshot here https://prnt.sc/ob9wzt

my screenshot of purchase https://prnt.sc/nz4lw6

I downloaded latest 3.9.1 pro version  and also tried  following
1)  redownload the nodejs 10.x from this page - https://nodejs.org
2) Then, went to the folder mstore-pro try with this script and start the simulator to run again:
watchman watch-del-all && rm -rf $TMPDIR/react-* && rm -rf node_modules/ && yarn cache clean && yarn install && yarn start -- --reset-cache
3) I also ran ./gradlew --warning-mode=all to get error warnings and got as below

> Configure project :react-native-webview
The DefaultSourceDirectorySet constructor has been deprecated. This is scheduled to be removed in Gradle 6.0. Please use the ObjectFactory service to create instances of SourceDirectorySet instead.
        at build_8r7uiyugh9o585njrygnvq6ck.run(/Users/chaitanya/Documents/mstore-3.9.1/mstore-pro/node_modules/react-native-webview/android/build.gradle:26)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
:react-native-webview:reactNativeAndroidRoot /Users/chaitanya/Documents/mstore-3.9.1/mstore-pro/node_modules/react-native/android

> Configure project :react-native-onesignal
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html
WARNING: Configuration 'testCompile' is obsolete and has been replaced with 'testImplementation'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Configure project :react-native-admob
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html
 
Please find build scan error from  https://scans.gradle.com/s/rf6brfx47taos

I also tried fresh installation of android studio and rpm install and so on steps even in windows separately  however I still get same error. so, it seems like there is been bug in your update of 3.9.1.

in previous tickets I was told to update latest version for fix of such errors however those errors remain same even in your latest updates.

I hope I get some permanent solution .

I am attaching file build.gradle file .

[build.gradle.txt](https://github.com/inspireui/support/files/3364396/build.gradle.txt)

**Comment by minhcasi** [comment]
Hi, it seems there is a problem when you changed the bundle id.

If your bundle ID is: touch.groceydude.in, then the folder format should be: java/com/touch/groceydude/in/MainApplication.java

Ref: https://docs.inspireui.com/mstore/delivery/

**Comment by smilechaitu** [comment]
Build successful now however it still shows some errors on screen as below:

Task :app:compileReleaseJavaWithJavac
Note: /Users/chaitanya/Documents/mstore-3.9.1/mstore-pro/android/app/src/main/java/com/touch/grocerydude/in/MainApplication.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/Users/chaitanya/.gradle/caches/transforms-2/files-2.1/caa62fb6d1985bd152a8f1d72799f4da/jars/classes.jar: D8: Type `com.google.android.gms.gcm.GoogleCloudMessaging` was not found, it is required for default or static interface methods desugaring of `java.lang.String com.onesignal.PushRegistratorGCM.getToken(java.lang.String)`
/Users/chaitanya/.gradle/caches/transforms-2/files-2.1/caa62fb6d1985bd152a8f1d72799f4da/jars/classes.jar: D8: Type `com.amazon.device.iap.model.ProductDataResponse$RequestStatus` was not found, it is required for default or static interface methods desugaring of `void com.onesignal.TrackAmazonPurchase$1.<clinit>()`
/Users/chaitanya/.gradle/caches/transforms-2/files-2.1/caa62fb6d1985bd152a8f1d72799f4da/jars/classes.jar: D8: Type `com.amazon.device.iap.PurchasingService` was not found, it is required for default or static interface methods desugaring of `void com.onesignal.TrackAmazonPurchase.setListener()`
/Users/chaitanya/.gradle/caches/transforms-2/files-2.1/caa62fb6d1985bd152a8f1d72799f4da/jars/classes.jar: D8: Interface `com.amazon.device.iap.PurchasingListener` not found. It's needed to make sure desugaring of `com.onesignal.TrackAmazonPurchase$OSPurchasingListener` is correct. Desugaring will assume that this interface has no default method.
/Users/chaitanya/.gradle/caches/modules-2/files-2.1/com.squareup.okhttp3/okhttp/3.12.1/dc6d02e4e68514eff5631963e28ca7742ac69efe/okhttp-3.12.1.jar: D8: Type `org.conscrypt.Conscrypt` was not found, it is required for default or static interface methods desugaring of `java.security.Provider okhttp3.internal.platform.ConscryptPlatform.getProvider()`

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1m 42s
298 actionable tasks: 150 executed, 148 up-to-date

Basically apk is generating but gives error as https://prnt.sc/oc2tjd when trying to install.

**Comment by smilechaitu** [comment]
hi, can you please update on issue?

I also found following issues
1) even though I changed Splash screen for some reason it still loads default M Store screen.
2) even though I changed App icon from in folder /mstore-3.9.1/mstore-pro/assets/icons it still shows your default icon of M store.

I also tried cache reset, again npm install and so on troubleshooting but no use. Apk is basically getting generated and using that Apk I can even install in mobile but it shows default m store app icon, splash screen and after that it crashes.  see screenshot here https://prnt.sc/odcxmc. 

I also checked AppConfig.json also config.js and both files got our company details not M Store details and in /mstore-pro/src/images folder also our images are there and default M- store images deleted too.

**Comment by minhcasi** [comment]
Hi, sorry for lated response:
To change splash screen on Expo version try to edit the app.json - https://docs.expo.io/versions/latest/guides/splash-screens/#__next

To change splash screen/icon on Pro version, you can refer to this guide - https://medium.com/the-react-native-log/change-default-launch-screen-in-react-native-ios-app-544f94f1e947

Let us know your checking - thank you!

**Comment by smilechaitu** [comment]
Hi,
   You did not read issue properly. guide which you sent is for IOS however issue I am facing is with Android app. I do not want to configure IOS app now as I want to launch only android App and also.

even though I placed icons, splash screen in relevant image folders it still pick ups default M store images and also Android app not working. kindly see this screenshot https://prnt.sc/odcxmc. so, these are two issues which I need solution. hope I get quick reply @minhcasi

**Comment by minhcasi** [comment]
Hi, you need to check the crash report from the google play store or install as service - https://sentry.io/welcome/ 

There are some missing config that causes the app crashing so you should debug it - thanks

**Comment by smilechaitu** [comment]
Here is error code. its same error coming even if I am deploying as default M store app without any changes in code.  it looks like code got bug?  @minhcasi 


com.facebook.react.common.JavascriptException: TypeError: TypeError: undefined is not an object (evaluating 'l.colors[0]')
This error is located at:
in l
in p
in withTheme(l)
in RCTView
in h
in AndroidHorizontalScrollContentView
in AndroidHorizontalScrollView
in u
in v
in f
in n
in p
in withTheme(n)
in n
in p
in withTheme(n)
in RCTView
in h
in RCTView
in RCTScrollView
in AndroidSwipeRefreshLayout
in c
in u
in v
in f
in l
in n
in p
in withTheme(n)
in withNavigation(withTheme(n))
in _
in RCTView
in n
in _
in p
in withTheme(Connect(n))
in RCTView
in o
in p
in withTheme(o)
in p
in RCTView
in RCTView
in RCTView
in l
in RCTView
in l
in n
in n
in P
in RCTView
in n
in RCTView
in l
in PanGestureHandler
in o
in L
in RCTView
in s
in o
in P
in v
in p
in RCTView
in RCTView
in l
in RCTView
in n
in RCTView
in n
in s
in P
in L
in RCTView
in RCTView
in l
in RCTView
in n
in n
in p
in withTheme(n)
in _
in p
in f
in _
in p
in withTheme(Connect(f))
in c
in s
in n
in RCTView
in RCTView
in c
This error is located at:
in L
in RCTView
in RCTView
in l
in RCTView
in n
in n
in p
in withTheme(n)
in _
in p
in f
in _
in p
in withTheme(Connect(f))
in c
in s
in n
in RCTView
in RCTView
in c, stack:
value@1307:1408
Wr@89:48199
ea@89:72359
ta@89:72849
Ma@89:80450
Da@89:79788
Ne@89:82845
Ie@89:13151
De@89:13324
receiveTouches@89:14083
value@31:3311
<unknown>@31:822
value@31:2565
value@31:794
value@-1
at com.facebook.react.modules.core.ExceptionsManagerModule.showOrThrowError(ExceptionsManagerModule.java
  at com.facebook.react.modules.core.ExceptionsManagerModule.reportFatalException (ExceptionsManagerModule.java:38)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:158)
  at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
  at android.os.Handler.handleCallback (Handler.java:873)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:29)
  at android.os.Looper.loop (Looper.java:193)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:232)
  at java.lang.Thread.run (Thread.java:764)

**Comment by minhcasi** [comment]
Hi, could you send your whole source code to support@inspireui.com - we will help to debug this issue, it seems you have an issue with the product variant - thanks

**Comment by smilechaitu** [comment]
Hi, I am not sure if its issue with product variant as Even with downloaded copy with default M-Store configuration which has default M store API  also its  giving same error however I sent email with full source code as well for investigation of this issue. 

Kindly check and revert back as soon as possible so, that we can launch app.

**Comment by smilechaitu** [comment]
any update on this bug ?? @concuacon

**Comment by minhcasi** [comment]
just note that we are checking, as the source has been modified so that will take us time for verify - thanks

**Comment by smilechaitu** [comment]
Hi,
    Please be noted source code was never modified. I just changed settings what's required to make app live such as replacing app names, icons and changing config as per your documentation thats all. 

This app apk worked in Version 3.8.0 if you use Expo and started giving errors for both expo and pro version after updates from 3.9.0 which shows code is been broken from your side.

also as mentioned in my previous messages I even tried using default settings of M-store with default M-store Wordpress API details. all I did was download latest version of code canyon and generated signing key and then generated into APK but even then I still see same error which means your app code has been broken. if you even test your latest original source code by generating new signing key you will know it easily 


I am trying make app live from 45 days and still could not make it due to always some errors 

Please tell how many more days it will take for you to check ?  if you think it will take another 10 days to check then its very hard to wait . @minhcasi

**Comment by minhcasi** [comment]
Hi @smilechaitu - kindly download the latest 3.9.3 to fix the issue - thank you!

**Comment by smilechaitu** [comment]
updated and it solved many issues however google shows app crashing for following devices
1) Pixel 3
2) Pixel 2
for android version 8.1 and 9.  

Issue: value@-1 and under this issue it shows following report

FATAL EXCEPTION: mqt_native_modules
Process: touch.grocerydude.in, PID: 12514
com.facebook.react.common.JavascriptException: Can't find variable: variation, stack:
<unknown>@1587:1439
forEach@-1
moveAllToCart@1587:1202
touchableHandlePress@176:3464
_performSideEffectsForTransition@182:9659
_performSideEffectsForTransition@-1
_receiveSignal@182:8341
_receiveSignal@-1
touchableHandleResponderRelease@182:5629
touchableHandleResponderRelease@-1
onResponderRelease@176:2574
E@82:1233
z@82:1376
U@82:1430
D@82:1721
H@82:2497
forEach@-1
j@82:2297
<unknown>@82:13831
Ue@82:81479
Me@82:13499
Fe@82:13672
receiveTouches@82:14441
value@26:3518
<unknown>@26:822
value@26:2772
value@26:794
value@-1
	at com.facebook.react.modules.core.ExceptionsManagerModule.showOrThrowError(ExceptionsManagerModule.java:54)
	at com.facebook.react.modules.core.ExceptionsManagerModule.reportFatalException(ExceptionsManagerModule.java:38)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:158)
	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
	at android.os.Looper.loop(Looper.java:193)
	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:232)
	at java.lang.Thread.run(Thread.java:764)
@minhcasi

**Comment by minhcasi** [comment]
Thank you, note for double-check on Pixel 3 & Pixel 2

**Comment by minhcasi** [comment]
Fixed on latest 3.9.4 - https://tppr.me/NktCI - thank you!
